src/actions/TemperatureSyncActions.js

```diff
@@ -226,6 +234,8 @@ const syncTemperatures = () => async (dispatch, getState) => {
   for (let i = 0; i < numberOfSensors; i++) {
     const sensor = sensors[i];
 
+    dispatch(updateSensorProgress(sensor));
```

Comment:
Out of curiousity, why not `forEach`?

```
const sensors = UIDatabase.objects('Sensor').filtered('location != null').slice();
...
sensors.forEach(sensor => {
  dispatch(updateSensorProgress(sensor));
  ...
}
```

Reply:
because of `await` the function would need to be `async` - considered adding an `asyncForEach` to the array prototype, thought that'd be fun but decided against :(

Reply:
Ah yeah, I've come up against that too. I believe can fix that by doing something like this (but maybe not too nice):

```
sensors.forEach(async sensor => ....)
```